src/ClassFileLocator.php

```diff
@@ -112,15 +112,16 @@ public function accept()
                             }
                             continue;
                         }
-                        list($type, $content, $line) = $token;
-                        switch ($type) {
-                            case T_STRING:
-                            case T_NS_SEPARATOR:
-                                $namespace .= $content;
-                                break;
+                        list($type, $content) = $token;
+                        $types = [T_STRING, T_NS_SEPARATOR];
+                        if (PHP_VERSION_ID >= 80000) {
+                            $types = [T_NAME_FULLY_QUALIFIED, T_NAME_QUALIFIED, T_STRING, T_NS_SEPARATOR];
```

Comment:
I would prefer having `array_merge` over re-assigning the all values as its easier to understand that PHP 8 just adds two new constants.

```suggestion
                            $types = array_merge($types, [T_NAME_FULLY_QUALIFIED, T_NAME_QUALIFIED]);
```

Reply:
that is a better way.